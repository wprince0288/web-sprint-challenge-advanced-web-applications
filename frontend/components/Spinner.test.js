// Import the Spinner component into this file and test
// that it renders what it should for the different props it can take.
import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Spinner from '../components/Spinner'

test('sanity', () => {
  expect(true).toBe(true)
});

describe('Spinner Componet', () => {
  it('renders nothing when "on" prop is false', () => {
    render(<Spinner on={false} />);
    expect(screen.queryByText(/please wait/i)).not.toBeInTheDocument();
  });

  test('renders the spinner when "on" prop is true', () => {
    render(<Spinner on={true} />);
    expect(screen.getByText(/please wait/i)).toBeInTheDocument();
    expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent('.');
    expect(screen.getByText(/please wait/i).closest('div')).toHaveAttribute('id', 'spinner');
});

test('matches snapshot when spinner is on', () => {
  const { asFragment } = render(<Spinner on={true} />);
  expect(asFragment()).toMatchSnapshot();
});

test('matches snapshot when spinner is off', () => {
  const { asFragment } = render(<Spinner on={false} />);
  expect(asFragment()).toMatchSnapshot();
});
  });

