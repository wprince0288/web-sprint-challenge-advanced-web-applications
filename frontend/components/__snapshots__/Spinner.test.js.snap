// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Componet matches snapshot when spinner is off 1`] = `<DocumentFragment />`;

exports[`Spinner Componet matches snapshot when spinner is on 1`] = `
<DocumentFragment>
  <div
    class="sc-beySPh cqHpCy"
    id="spinner"
  >
    <h3>
       .
    </h3>
       Please wait...
  </div>
</DocumentFragment>
`;
